gtdynamics/optimizer/ConstraintManifold.cpp

```diff
@@ -0,0 +1,275 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  ConstraintManifold.cpp
+ * @brief Constraint manifold implementations.
+ * @author: Yetong Zhang
+ */
+
+#include <gtdynamics/optimizer/ConstraintManifold.h>
+#include <gtdynamics/optimizer/AugmentedLagrangianOptimizer.h>
+#include <gtdynamics/optimizer/PenaltyMethodOptimizer.h>
+
+#include <gtsam/linear/GaussianBayesNet.h>
+#include <gtsam/linear/VectorValues.h>
+#include <gtsam/nonlinear/LinearContainerFactor.h>
+
+namespace gtsam {
+
+/* ************************************************************************* */
+void ConstraintManifold::compute_values(const gtsam::Values& values,
+                                        const bool retract_init) {
+  values_ = gtsam::Values();
```

Comment:
Any reason for new memory initialization rather than just clearing `values_` aka `values_.clear()`?

Reply:
Is there a difference in complexity between the two ways?

Reply:
Nope, there shouldn't be.